tests/e2e-pw/specs/merchant/merchant-orders-partial-refund.spec.ts

```diff
@@ -0,0 +1,178 @@
+
+/**
+ * Internal dependencies
+ */
+import { config } from '../../config/default';
+import { getMerchant, getShopper } from '../../utils/helpers';
+import {
+	fillCardDetails,
+	placeOrder,
+	setupProductCheckout,
+} from '../../utils/shopper';
+import { goToShop } from '../../utils/shopper-navigation';
+import { goToOrder } from '../../utils/merchant-navigation';
+import {
+	activateMulticurrency,
+	deactivateMulticurrency,
+	restoreCurrencies,
+} from '../../utils/merchant';
+
+// Needs to be finished.
+test.describe( 'Order > Partial refund', () => {
+	const product1 = config.products.simple.name;
+	const product2 = 'Belt';
+	const product3 = 'Hoodie with Logo';
+
+	/**
+	 * Elements:
+	 * - test title
+	 * - object containing the items to be ordered, and the quantities and amounts to be refunded
+	 */
+	const dataTable: [
+		string,
+		{
+			lineItems: Array< [ string, number ] >;
+			refundInputs: { refundQty: number; refundAmount: number }[];
+		}
+	][] = [
+		[
+			'Partially refund one product of two product order',
+			{
+				lineItems: [
+					[ product1, 1 ],
+					[ product2, 1 ],
+				],
+				refundInputs: [ { refundQty: 0, refundAmount: 5 } ],
+			},
+		],
+		[
+			'Refund two products of three product order',
+			{
+				lineItems: [
+					[ product1, 1 ],
+					[ product2, 2 ],
+					[ product3, 1 ],
+				],
+				refundInputs: [
+					{ refundQty: 1, refundAmount: 18 },
+					{ refundQty: 1, refundAmount: 55 },
+				],
+			},
+		],
+	];
+
+	let firstOrderId: string;
+	let secondOrderId: string;
+	let orderIds: string[];
+	let orderTotal: string;
+	let wasMulticurrencyEnabled = false;
+	let merchantPage: Page, shopperPage: Page;
+
+	const orderProducts = async ( { browser }, dataTableIndex: number ) => {
+		shopperPage = ( await getShopper( browser ) ).shopperPage;
+		const lineItems = dataTable[ dataTableIndex ][ 1 ].lineItems;
+		await goToShop( shopperPage );
+		await setupProductCheckout( shopperPage, lineItems );
+		await fillCardDetails( shopperPage );
+		await placeOrder( shopperPage );
+		await expect(
+			shopperPage.getByRole( 'heading', { name: 'Order received' } )
+		).toBeVisible();
+
+		const orderIdField = shopperPage.locator(
+			'.woocommerce-order-overview__order.order > strong'
+		);
+		const orderId = await orderIdField.innerText();
+
+		return orderId;
+	};
+
+	test.beforeAll( async ( { browser } ) => {
+		merchantPage = ( await getMerchant( browser ) ).merchantPage;
+		wasMulticurrencyEnabled = await activateMulticurrency( merchantPage );
+		await restoreCurrencies( merchantPage );
+		firstOrderId = await orderProducts( { browser }, 0 );
+		secondOrderId = await orderProducts( { browser }, 1 );
+		orderIds = [ firstOrderId, secondOrderId ];
+	} );
+
+	test.afterAll( async () => {
+		if ( ! wasMulticurrencyEnabled ) {
+			await deactivateMulticurrency( merchantPage );
+		}
+	} );
+
+	for ( let i = 0; i < dataTable.length; i++ ) {
+		test( dataTable[ i ][ 0 ], async ( { browser } ) => {
```

Comment:
Nitpick: The test name is missing the `should refund ${testName}` prefix here.

Reply:
I didn't include the `should refund` prefix here as I renamed the tests to be more descriptive.

In the Puppeteer spec they were named:

- `a single line item`
- `several line items`

I changed them to:

- `Partially refund one product of two product order`
- `Refund two products of three product order`

And that is how they are displayed when the tests are running:

```
[3/4] …rchant/merchant-orders-partial-refund.spec.ts:108:7 › Order > Partial refund › Partially refund one product of two product order

[4/4] …t] › merchant/merchant-orders-partial-refund.spec.ts:108:7 › Order > Partial refund › Refund two products of three product order
```

Do you think that is sufficient?